src/common/mfu_daos.c

```diff
@@ -1041,6 +1063,20 @@ int daos_connect(
              * create a POSIX container. Otherwise, create a container of
              * the same type as the source. */
             if (!have_src_cont || (da->src_cont_type == DAOS_PROP_CO_LAYOUT_POSIX)) {
+#ifdef HDF5_SUPPORT
+                /* read container properties in hdf5 file when moving data
+                 * back to DAOS if the preserve option has been set */
+                if (preserve) {
+                    /* TODO: should always be called daos_metadata.hdf5? */
```

Comment:
@dsikich I think the simplest approach is for the user to provide the filename. But if we want to get "smart" about it, we could:
1. If daos_preserve_path is a directory, look for daos_metadata.hdf5
2. If daos_perserve_path is a file, use the file.

We could do this both for serializing and deserializing the metadata; though, I'm not sure if we want that complexity

Reply:
@daltonbohning ok, yeah it is probably simplest to just expect the user to provide a filename

Reply:
@dsikich It might be worth asking about this in the datamover WG to get other opinions